packages/core/src/fhir-deduplication/resources/allergy-intolerance.ts

```diff
@@ -0,0 +1,41 @@
+import { AllergyIntolerance } from "@medplum/fhirtypes";
+import { combineResources, fillMaps } from "../shared";
+
+export function deduplicateAllergyIntolerances(allergies: AllergyIntolerance[]) {
+  const { allergiesMap, refReplacementMap } = groupSameAllergies(allergies);
+  return {
+    combinedAllergies: combineResources({
+      combinedMaps: [allergiesMap],
+    }),
+    refReplacementMap,
+  };
+}
+
+/**
+ * Approach:
+ * 1 map, where the key is made of:
+ * - reaction substance
+ *
+ * We want to remove "unknown" substances and manifestations in the process
+ */
+export function groupSameAllergies(allergies: AllergyIntolerance[]): {
+  allergiesMap: Map<string, AllergyIntolerance>;
+  refReplacementMap: Map<string, string[]>;
+} {
+  const allergiesMap = new Map<string, AllergyIntolerance>();
+  const refReplacementMap = new Map<string, string[]>();
+
+  for (const allergy of allergies) {
+    const reaction = allergy.reaction;
+
+    if (reaction) {
```

Comment:
just a reminder here ( i know this is wip) about 
1. removing unknown allergies and filtering out unknown substances and manifestations
2. only using the substance as the key